libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1614,15 +1620,17 @@ bool TypeChecker::visit(UnaryOperation const& _operation)
 	else
 		_operation.subExpression().accept(*this);
 	TypePointer const& subExprType = type(_operation.subExpression());
-	TypePointer t = type(_operation.subExpression())->unaryOperatorResult(op);
+	TypeResult result = type(_operation.subExpression())->unaryOperatorResult(op);
+	TypePointer t = TypePointer(result);
```

Comment:
It might be better readable to use a function of result here, but I'm not sure.

Reply:
The current interface also allows ``TypePointer t = result.get()``. Would you prefer to use it like that? Ultimately, I'd try to reduce those implicit conversions to a minimum (which would need further refactoring in ``Types.cpp`` and probably structured bindings :wink:)